packages/light.js/src/rpc/other/makeContract.ts

```diff
@@ -29,35 +29,37 @@ interface MakeContract {
  *
  * @param address - The contract address.
  * @param abiJson - The contract abi.
+ * @param api - The api Object.
  * @return - The contract object as defined in @parity/api.
  */
 const getContract = memoizee(
-  (address: Address, abiJson: any[], provider: any) => {
-    const api = provider ? createApiFromProvider(provider) : getApi();
-    return api.newContract(abiJson, address);
-  }, // use types from @parity/abi
+  (address: Address, abiJson: any[], api: any) =>
+    api.newContract(abiJson, address), // use types from @parity/abi
   { length: 1 } // Only memoize by address
```

Comment:
does the memoization here needs to be based on api also or not?

Reply:
Yes, `api.newContract(...).contractMethod.call()` makes a rpc call with that api object.